scripts/install/providers/github/github.sh

```diff
@@ -65,15 +73,160 @@ function configure_repo {
   _debug "$CODE_REPO_GIT_HTTP_URL"
   _debug "$CODE_REPO_ID"
 
-  # Configure remote for local git repo
-  remoteWithCreds="https://${GH_PAT}@github.com/${GH_ORG_NAME}/${GH_Repo_NAME}.git"
-  git init
-  git branch -m main
-  git remote add origin "$remoteWithCreds"
-
   _success "Repo '${GH_Repo_NAME}' created."
 }
 
+function configure_runners {
+  # Replace in all files in ./.github/workflows "runs-on: ubuntu-latest" with "runs-on: self-hosted"
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+      for file in ./.github/workflows/*; do
+          sed -i '' -E 's/([[:space:]]*)runs-on: ubuntu-latest([[:space:]]*)/\1runs-on: self-hosted\2/g' "$file"
+      done
+  else
+      for file in ./.github/workflows/*; do
+          sed -i -E 's/([[:space:]]*)runs-on: ubuntu-latest([[:space:]]*)/\1runs-on: self-hosted\2/g' "$file"
+      done
+  fi
+
+  # Verify that required variables exist
+  local REQUIRED_VARS=(\
+      GH_ORG_NAME \
+      GH_Repo_NAME \
+      RUNNERS_RESOURCE_GROUP \
+      RUNNERS_LOCATION \
+      RUNNERS_SUBNET \
+      RUNNERS_PUBLIC_KEY_PATH)
+  for var in "${REQUIRED_VARS[@]}"; do
+      if [[ -z "${!var}" ]]; then
+          _error "The variable '$var' is not defined. Aborting."
+          exit 1
+      fi
+  done
+
+  # Validate that the file exists
+  if [[ ! -f "$RUNNERS_PUBLIC_KEY_PATH" ]]; then
+      _error "The specified public key file '$RUNNERS_PUBLIC_KEY_PATH' does not exist. Aborting."
+      exit 1
+  fi
+
+  # Optional parameter: VM size (default: Standard_D2s_v3)
+  RUNNERS_AZURE_VM_SIZE=${RUNNERS_AZURE_VM_SIZE:-Standard_D2s_v3}
+
+  # Optional parameter: Number of runners to launch (default: 1)
+  RUNNERS_COUNT=${RUNNERS_COUNT:-1}
+
+  # Optional parameter: Image (default: Ubuntu2404)
+  RUNNERS_AZURE_IMAGE=${RUNNERS_AZURE_IMAGE:-Ubuntu2404}
+
+  # Optional parameter: VM name (default: symphony-github-runner)
+  RUNNERS_VM_NAME=${RUNNERS_VM_NAME:-symphony-github-runner}-${GH_Repo_NAME}
+
+  # Optional parameter: VM username (default: azureuser)
+  RUNNERS_VM_USERNAME=${RUNNERS_VM_USERNAME:-azureuser}
+
+  # Retrieve multiple GitHub runner registration tokens
+  _information "Retrieving ${RUNNERS_COUNT} GitHub runner registration tokens..."
+  local tokens=()
+
+  for i in $(seq 1 "$RUNNERS_COUNT"); do
+      local token=$(gh api -X POST /repos/${GH_ORG_NAME}/${GH_Repo_NAME}/actions/runners/registration-token -q .token)
+
+      if [[ "$token" == "null" || -z "$token" ]]; then
+          _error "Error: Failed to retrieve token for runner $i."
+          exit 1
+      fi
+
+      tokens+=("$token")
+
+      _information "Sleeping for 1 second to avoid repeated token requests..."
+      sleep 1
+  done
+
+  _information "All tokens retrieved successfully."
+
+  # Convert tokens array into a space-separated string
+  local token_string=$(IFS=" " ; echo "${tokens[*]}")
+
+  # Generate the cloud-init file (cloud-init.yaml)
+  cat > cloud-init.yaml <<EOF
```

Comment:
nit: consider reading the cloud init template from a file in the templates folder. This will make it easier for future changes without having to go into bash.

Behavior would be the same, just thinking of maintainability.